src/SprykerEco/Zed/PunchoutCatalogs/Communication/Controller/CompanyBusinessUnitController.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+/**
+ * Copyright © 2016-present Spryker Systems GmbH. All rights reserved.
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+namespace SprykerEco\Zed\PunchoutCatalogs\Communication\Controller;
+
+use Generated\Shared\Transfer\CompanyBusinessUnitCollectionTransfer;
+use Generated\Shared\Transfer\CompanyBusinessUnitCriteriaFilterTransfer;
+use Generated\Shared\Transfer\CompanyBusinessUnitTransfer;
+use Spryker\Zed\Kernel\Communication\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+
+/**
+ * @method \SprykerEco\Zed\PunchoutCatalogs\Communication\PunchoutCatalogsCommunicationFactory getFactory()
+ * @method \SprykerEco\Zed\PunchoutCatalogs\Business\PunchoutCatalogsFacadeInterface getFacade()
+ * @method \SprykerEco\Zed\PunchoutCatalogs\Persistence\PunchoutCatalogsRepositoryInterface getRepository()
+ */
+class CompanyBusinessUnitController extends AbstractController
+{
+    protected const KEY_ID = 'id';
+    protected const KEY_TEXT = 'text';
+    protected const KEY_RESULTS = 'results';
+
+    protected const PARAM_PARENT_COMPANY_BUSINESS_UNIT = 'id-parent-company-business-unit';
+
+    /**
+     * @param \Symfony\Component\HttpFoundation\Request $request
+     *
+     * @throws \Symfony\Component\HttpKernel\Exception\NotFoundHttpException
+     *
+     * @return \Symfony\Component\HttpFoundation\JsonResponse
+     */
+    public function indexAction(Request $request): JsonResponse
+    {
+        $idParentCompanyBusinessUnit = $this->castId($request->query->getInt(static::PARAM_PARENT_COMPANY_BUSINESS_UNIT));
```

Comment:
Maybe it would be better to move `$request->query->getInt(static::PARAM_PARENT_COMPANY_BUSINESS_UNIT)` to the next line?